04.ls/ls.rb

```diff
@@ -33,4 +33,113 @@ def split_files(files)
   divided_files
 end
 
+PERMISSION = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx',
+  'fifo' => 'p',
+  'characterSpecial' => 'c',
+  'directory' => 'd',
+  'blockSpecial' => 'b',
+  'file' => '-',
+  'link' => 'l',
+  'socket' => 's'
+}.freeze
+
+def load_l_option(path, file_names)
+  stat_files = file_names.map { |faile_name| File.lstat("#{path}/#{faile_name}") }
+  path_to_stat = stat_files.map.with_index { |stat, idx| ["#{path}/#{file_names.fetch(idx)}", stat] }.to_h
+
+  puts "total #{stat_files.map(&:blocks).sum}"
+  load_body(path_to_stat).each do |file|
+    puts format_file(file, stat_files)
+  end
+end
+
+def load_body(path_to_stat)
+  path_to_stat.map do |absolute_path, stat|
+    {
+      permission: to_permission(stat),
+      link_count: stat.nlink.to_s,
+      owner: Etc.getpwuid(stat.uid).name,
+      group: Etc.getgrgid(stat.gid).name,
+      byte_size: stat.size.to_s,
+      time_stamp: stat.mtime.strftime(
+        Time.now.year == stat.mtime.year ? ' %_m %_d %H:%M ' : ' %_m %_d  %Y '
+      ),
+      file_name:
+        if stat.symlink?
+          "#{File.basename(absolute_path)} -> #{File.readlink(absolute_path)}"
+        else
+          File.basename(absolute_path)
+        end
+    }
+  end
+end
+
+def format_file(file, stat_files)
+  block_to_maxlength = fetch_max_length(stat_files)
+  constant_size = 2
+  [
+    file.fetch(:permission),
+    file.fetch(:link_count).rjust(block_to_maxlength.fetch(:link) + constant_size), ' ',
+    file.fetch(:owner).ljust(block_to_maxlength.fetch(:owner)     + constant_size),
+    file.fetch(:group).ljust(block_to_maxlength.fetch(:group)     + constant_size),
+    file.fetch(:byte_size).rjust(block_to_maxlength.fetch(:byte_size)),
+    file.fetch(:time_stamp),
+    file.fetch(:file_name)
+  ].join
+end
+
+def fetch_max_length(stat_files)
+  block_to_all_value = { link: [], owner: [], group: [], byte_size: [] }
+  stat_files.each do |stat|
+    block_to_all_value.fetch(:link)     << stat.nlink
```

Comment:
```suggestion
  stat_files.each do |stat|
    sleep 0.2
    block_to_all_value.fetch(:link)     << stat.nlink
```

こんな感じでsleepさせるとめちゃくちゃ遅くなりませんか？
非効率なロジックになってないかロジック全体を見直してみてください〜。

Reply:
1. ハッシュとして各ブロックに初期値をセット。
1. 各ブロックの値を取得、文字数を確認　ハッシュの値より大きければ入れ替える。
2. 以降繰り返し。
といったロジックに修正しました。